File: AN_2/SEMESTRUL_IV/MPP/LABORATOR/JAVA_TEMA/tema_bun/Services/src/main/java/com/example/tema_bun/IServices.java
package com.example.tema_bun;

import java.io.IOException;
import java.util.List;

public interface IServices {
    boolean login(String username, String parola , IObserver client) throws Exception;

    void logout(String email) throws Exception;

    Utilizator getByAccount(String username,String parola);

    void createAccount(String username,String parola) throws IOException, Exception;

    List<Integer> creareVectorCnt() throws Exception;

    void adaugaCopilInscriere(String nume, Integer varsta, List<String> probe,String usernameCurent) throws Exception;

    List<String> JoinInscrieriCopii(String proba,String categorie) throws Exception;
}

File: AN_2/SEMESTRUL_IV/MPP/LABORATOR/CSHARP_TEMA/tema_sapt3/Services/IServices.cs
using Model;

namespace services;

public interface IServices
{
    bool login(string username, string parola, IObserver client);

    void logout(string username);
    Utilizator getByAccount(string username,string parola);
    void createAccount(String username, String parola);
    List<int> creareVectorCnt();
    List<int> gasesteParticipanti(String proba, String categorie);
    String getNumeVarstaById(int id);
    List<String> gasesteNumeVarstaCopii(List<int> listCoduri);
    void adaugaCopilInscriere(String nume, int varsta, List<String> probe,String usernameCurent);
    List<string> JoinInscrieriCopii(String proba, String categorie);

    //REST
    Proba addProba(Proba proba);
    Proba deleteProba(Proba proba);
    List<Proba> getAllProba();
    Proba findByIdProba(String id_proba);
}